### PR TITLE
Add correct answers and explanation.

### PR DESCRIPTION
**Pull request by szirbucz**

### CODE REVIEW

File: src/tasks [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
 (Multiple choice)
-* Single Responsibility Principle (yes)
-* Open-Close principle (yes)
-* Liskov Substitution Principle (no)
-* Inversion of Control (no, tricky, since it's not a SOLID principle :) )
-* Dependency Inversion Principle (yes) 
+* Single Responsibility Principle (yes) [There are multiple reason to modify: print, order, composition]
+* Open-Close principle (yes) [PrintingService is hard-wired, we can't extend it without modifying the NameService]
+* Liskov Substitution Principle (no) [No inheritance]
+* Inversion of Control (no) [Tricky, since it's not a SOLID principle :)]
```
Comment (magwas) [inline src/tasks:6]: but otherwise violated?

File: src/tasks [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
+* Dependency Inversion Principle (yes) [NameService depends on lower level PrintingService] 
 
 2. Which principle is violated here and why?
 
-* Open-Close Principle, because Square is not Open for extension.
-* Single Responsibility, because Rectangle and Square have the same responsibility
-* Liskov Substitution Principle, because Square can't be used instead Rectangle. (helyes)
-* Interface Segregation Principle, because there is no Interface implemented.
-* Dependency Inversion Principle, because the lowest level Square depends on Rectangle.
+* Open-Close Principle, because Square is not Open for extension. (no)
```
Comment (magwas) [inline src/tasks:11]: Make the format uniform. Above there was answer and reasoning in different brackets. I like that.

File: src/tasks [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
-* Each Shape implementation violates Single Responsibility Principle, because they calculate area and perimeter too.
+* No violation is present. (no)
+* The Shape interface violates the Interface Segregation Principle, since it contains more than one method. (no) 
+* Circle and Rectangle implements more than one interface, it violates the Interface Segregation Principle. (no)
+* Shape contains method declaration that not applicable for all shapes, that violates the Interface Segregation Principle. (yes)
+* Each Shape implementation violates Single Responsibility Principle, because they calculate area and perimeter too. (no)
 
 4. What rules are violated in the getLeadersOfCompany method?
 (Multiple choice)
 * Interface Segregation Principle (no)
```
Comment (magwas) [inline src/tasks]: Maybe the NOs should also be explained.

**Review by magwas — CHANGES_REQUESTED** [review]

File: src/tasks (hunk 1 above)
Comment (szirbucz) [reply src/tasks:6]: Yes. The PrintingService gets the control in procedural way.

File: src/tasks (hunk 2 above)
Comment (szirbucz) [reply src/tasks:11]: But reasoning is included to the option here.

**Review by magwas — APPROVED** [review]

**Pull request merged**